util/topsql/topsql.go

```diff
@@ -36,14 +40,32 @@ const (
 	MaxPlanTextSize = 32 * 1024
 )
 
-var globalTopSQLReport reporter.TopSQLReporter
+var (
+	globalTopSQLReport        reporter.TopSQLReporter
+	globalTopSQLPubSubService tipb.TopSQLPubSubServer
+)
 
 // SetupTopSQL sets up the top-sql worker.
 func SetupTopSQL() {
 	rc := reporter.NewGRPCReportClient(plancodec.DecodeNormalizedPlan)
-	globalTopSQLReport = reporter.NewRemoteTopSQLReporter(rc)
+	cr := reporter.NewReportClientRegistry()
+	globalTopSQLReport = reporter.NewRemoteTopSQLReporter(cr, rc)
 	tracecpu.GlobalSQLCPUProfiler.SetCollector(globalTopSQLReport)
 	tracecpu.GlobalSQLCPUProfiler.Run()
+
+	publisher := reporter.NewTopSQLPublisher(plancodec.DecodeNormalizedPlan, cr)
+	globalTopSQLPubSubService = publisher
+
+	if len(config.GetGlobalConfig().TopSQL.ReceiverAddress) != 0 {
```

Comment:
We can remove config `TopSQL.ReceiverAddress` now?

Reply:
For compatibility, I prefer not to remove it.